include/nbl/builtin/hlsl/macros.hlsl

```diff
@@ -0,0 +1,35 @@
+// Copyright (C) 2022 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+#ifndef _NBL_BUILTIN_HLSL_MACROS_INCLUDED_
+#define _NBL_BUILTIN_HLSL_MACROS_INCLUDED_
+
+#define NBL_HLSL_EVAL(X) X
+
+#define NBL_HLSL_EQUAL(X,Y) (NBL_HLSL_EVAL(X)==NBL_HLSL_EVAL(Y))
+#define NBL_HLSL_NOT_EQUAL(X,Y) (NBL_HLSL_EVAL(X)!=NBL_HLSL_EVAL(Y))
+
+#define NBL_HLSL_LESS(X,Y) (NBL_HLSL_EVAL(X)<NBL_HLSL_EVAL(Y))
+#define NBL_HLSL_GREATER(X,Y) (NBL_HLSL_EVAL(X)>NBL_HLSL_EVAL(Y))
+
+#define NBL_HLSL_IS_NOT_POT(v) (NBL_HLSL_EVAL(v)&(NBL_HLSL_EVAL(v)-1))!=0
+#define NBL_HLSL_IS_POT(v) (!NBL_HLSL_IS_NOT_POT(v))
+
+#define NBL_HLSL_CONCAT_IMPL2(X,Y) X ## Y
+#define NBL_HLSL_CONCAT_IMPL(X,Y) NBL_HLSL_CONCAT_IMPL2(X,Y)
+#define NBL_HLSL_CONCATENATE2(X,Y) NBL_HLSL_CONCAT_IMPL(NBL_HLSL_EVAL(X),NBL_HLSL_EVAL(Y))
+#define NBL_HLSL_CONCATENATE3(X,Y,Z) NBL_HLSL_CONCATENATE2(NBL_HLSL_CONCATENATE2(X,Y),NBL_HLSL_EVAL(Z))
+#define NBL_HLSL_CONCATENATE4(X,Y,Z,W) NBL_HLSL_CONCATENATE2(NBL_HLSL_CONCATENATE2(X,Y),NBL_HLSL_CONCATENATE2(Z,W))
+
+#define NBL_HLSL_AND(X,Y) (NBL_HLSL_EVAL(X)&NBL_HLSL_EVAL(Y))
+#define NBL_HLSL_OR(X,Y) (NBL_HLSL_EVAL(X)|NBL_HLSL_EVAL(Y))
+#define NBL_HLSL_XOR(X,Y) (NBL_HLSL_EVAL(X)^NBL_HLSL_EVAL(Y))
+
+#define NBL_HLSL_ADD(X,Y) (NBL_HLSL_EVAL(X)+NBL_HLSL_EVAL(Y))
+#define NBL_HLSL_SUB(X,Y) (NBL_HLSL_EVAL(X)-NBL_HLSL_EVAL(Y))
+#define NBL_HLSL_MUL(X,Y) (NBL_HLSL_EVAL(X)*NBL_HLSL_EVAL(Y))
+
+#define NBL_HLSL_MAX(X,Y) (((NBL_HLSL_EVAL(X))>(NBL_HLSL_EVAL(Y))) ? (NBL_HLSL_EVAL(X)):(NBL_HLSL_EVAL(Y)))
+#define NBL_HLSL_MIN(X,Y) (((NBL_HLSL_EVAL(X))<(NBL_HLSL_EVAL(Y))) ? (NBL_HLSL_EVAL(X)):(NBL_HLSL_EVAL(Y)))
```

Comment:
@AnastaZIuk unify with `nbl/macros.h` (or move this and parts of `nbl/macros.h` to `nbl/builtin/macros.h`)

Reply:
@atkurtul's task now

Reply:
@kpentaris when #546 gets merged all the NBL_HLSL_ here will go away as they can be done with templates, except for `NBL_HLSL_EVAL` which will become `NBL_EVAL`